### ISSUE TITLE
OCI Image support

### ISSUE THREAD

**Issue by naterini** [hpc/charliecloud#1256]
The current charliecloud implementation for OCI images has a good number of hard dependencies for podman. It would be nice if charliecloud could have generic support for OCI images instead.

**Comment by sc13-bioinf** [comment]
So I wrote the following fragment to make a version of ch-builder2tar work for podman. It seemed to work ok. Should I try and add this to the ch-builder2tar script?

ch-podman2tar
#!/usr/bin/env bash



while true; do
    case $1 in
        -b|--builder)
            shift
            CH_BUILDER=$1
            shift
            ;;
        --nocompress)
            nocompress=yes
            shift
            ;;
        *)
            break
            ;;
        esac
done

# NOTE: This is used both to name user-visible stuff like tarballs as well as
# dig around in the ch-image storage directory.
tag_to_path () {
    echo "$1" | sed 's/\//%/g'
}

# sc13 we only need the docker bit
numArgs=$#


if [[ $numArgs -gt 1 ]]; then

image=$1
shift

outdir=$1
shift

tar=${outdir}/$(tag_to_path "$image").tar
tar_gzipped=${tar}.gz

echo "tar: '${tar}'"
echo "tgz: '${tar_gzipped}'"

# Export the image to tarball.
echo "exporting" 
cid=$(podman container create --read-only "$image")
echo "obtained cid: '${cid}'"
#size=$(docker image inspect "$image" --format='{{.Size}}')
podman export --output "$tar" "${cid}"
podman rm "$cid" > /dev/null

echo "adding environment"
temp=$(mktemp -t ch-builder2tar.XXXXXX)
echo "temp is: '${temp}'"
podman image inspect --format='{{range .Config.Env}}{{println .}}{{end}}' "$image" > "$temp"
gnutar rf "$tar" -b1 -P --xform="s|${temp}|ch/environment|" "$temp"
rm "$temp"

cat < "$tar" | gzip -6 > "$tar_gzipped"
rm "$tar"
else
echo "Please supply image name and dest dir"
exit 1
fi

**Comment by reidpr** [comment]
Absolutely! That's fantastic. I'd love a PR for that. `ch-builder2tar` is deprecated though, so the changes should go in `ch-convert`.
